Gems/Terrain/Assets/Shaders/Terrain/TerrainDetailHelpers.azsli

```diff
@@ -489,22 +487,22 @@ bool GetDetailSurface(inout DetailSurface surface, float2 detailMaterialIdCoord,
             }
 
             WeightDetailSurface(surface, 1.0 - mat2Weight);
-            GetDetailSurfaceForMaterial(tempSurface, GetMaterialContext(material2.x, detailUv), macroColor);
+            GetDetailSurfaceForMaterial(tempSurface, GetMaterialContext(material2.x, detailUv, detailUvDdx, detailUvDdy), macroColor);
             WeightDetailSurface(tempSurface, mat2Weight);
             AddDetailSurface(surface, tempSurface);
         }
     }
     else
     {
         MaterialContext materialContexts[8] = {
-            GetMaterialContext(material1.x, detailUv),
-            GetMaterialContext(material2.x, detailUv),
-            GetMaterialContext(material1.y, detailUv),
-            GetMaterialContext(material2.y, detailUv),
-            GetMaterialContext(material1.z, detailUv),
-            GetMaterialContext(material2.z, detailUv),
-            GetMaterialContext(material1.w, detailUv),
-            GetMaterialContext(material2.w, detailUv)
+            GetMaterialContext(material1.x, detailUv, detailUvDdx, detailUvDdy),
```

Comment:
Nit: from a variable naming perspective, something like `detaildUVdx` is a bit more natural since the first "d" refers to the differential of `Uv` (which is taken with respect to `dx` or `dy`).

Reply:
Yeah.. But it's hard to handle camel case. We also have ddx(position), writing dPositiondx seems a bit odd, positionDdx feels better to me.